controllers/workspace/predicates.go

```diff
@@ -47,3 +51,31 @@ var predicates = predicate.Funcs{
 	},
 	GenericFunc: func(_ event.GenericEvent) bool { return true },
 }
+
+var podPredicate = predicate.Funcs{
+	UpdateFunc: func(ev event.UpdateEvent) bool {
+		newObj, ok := ev.ObjectNew.(*corev1.Pod)
+		if !ok {
+			return true
+		}
+
+		oldObj, ok := ev.ObjectOld.(*corev1.Pod)
+		if !ok {
+			// Should never happen
+			return true
+		}
+
+		// If the dewworkspace label does not exist, do no reconcile
+		if _, ok := newObj.Labels[constants.DevWorkspaceIDLabel]; !ok {
```

Comment:
If there's no devworkspace name label, then we don't enqueue any reconciles and so the functionality is the same. We should be checking workspace ID here since it's the canonical label that identifies resources are related to DevWorkspaces.

Reply:
True, we could drop it earlier but it does not matter since we must provide both labels.